tests/integration/operatorhub/cmd_service_test.go

```diff
@@ -29,521 +29,427 @@ var _ = Describe("odo service command tests for OperatorHub", func() {
+				expected := []string{"EtcdCluster", "etcd Cluster"}
+				Expect(helper.GjsonMatcher(values, expected)).To(Equal(true))
 			})
 
-			helper.CmdShouldPass("odo", "service", "delete", "EtcdCluster/example", "-f")
-		})
-
-		It("should be able to create a service with name passed on CLI", func() {
-			helper.CmdShouldPass("odo", "create", "nodejs")
-
-			name := helper.RandString(6)
-			// First let's grab the etcd operator's name from "odo catalog list services" output
-			operators := helper.CmdShouldPass("odo", "catalog", "list", "services")
-			etcdOperator := regexp.MustCompile(`etcdoperator\.*[a-z][0-9]\.[0-9]\.[0-9]-clusterwide`).FindString(operators)
-
-			stdOut := helper.CmdShouldPass("odo", "service", "create", fmt.Sprintf("%s/EtcdCluster", etcdOperator), "--dry-run", "--project", commonVar.Project)
+			It("should find the services by keyword", func() {
+				stdOut := helper.CmdShouldPass("odo", "catalog", "search", "service", "etcd")
+				helper.MatchAllInOutput(stdOut, []string{"etcdoperator", "EtcdCluster"})
 
-			// stdOut contains the yaml specification. Store it to a file
-			randomFileName := helper.RandString(6) + ".yaml"
-			fileName := filepath.Join(os.TempDir(), randomFileName)
-			if err := ioutil.WriteFile(fileName, []byte(stdOut), 0644); err != nil {
-				fmt.Printf("Could not write yaml spec to file %s because of the error %v", fileName, err.Error())
-			}
-
-			// now create operator backed service
-			helper.CmdShouldPass("odo", "service", "create", "--from-file", fileName, name, "--project", commonVar.Project)
-			helper.CmdShouldPass("odo", "push")
+				stdOut = helper.CmdShouldPass("odo", "catalog", "search", "service", "EtcdCluster")
+				helper.MatchAllInOutput(stdOut, []string{"etcdoperator", "EtcdCluster"})
 
-			// Attempting to create service with same name should fail
-			stdOut = helper.CmdShouldFail("odo", "service", "create", "--from-file", fileName, name, "--project", commonVar.Project)
-			Expect(stdOut).To(ContainSubstring("please provide a different name or delete the existing service first"))
-		})
-	})
-
-	Context("When using from-file option", func() {
-
-		var tmpContext string
-
-		JustBeforeEach(func() {
-			tmpContext = helper.CreateNewContext()
-			preSetup()
-		})
+				stdOut = helper.CmdShouldFail("odo", "catalog", "search", "service", "dummy")
+				Expect(stdOut).To(ContainSubstring("no service matched the query: dummy"))
+			})
 
-		JustAfterEach(func() {
-			helper.DeleteDir(tmpContext)
-			cleanPreSetup()
-		})
+			It("should list the operator in JSON output", func() {
+				jsonOut := helper.CmdShouldPass("odo", "catalog", "list", "services", "-o", "json")
+				helper.MatchAllInOutput(jsonOut, []string{"etcdoperator"})
+			})
 
-		It("should fail to create service if metadata doesn't exist or is invalid", func() {
-			helper.CmdShouldPass("odo", "create", "nodejs")
-			noMetadata := `
+			When("a nodejs component is created", func() {
```

Comment:
I am nitpicking, but can you also make this a generic description - a ~nodejs~ component is created?